src/app/dashboard/settings/settings.component.ts

```diff
@@ -0,0 +1,116 @@
+import { CommonModule } from '@angular/common';
+import { Component, OnInit } from '@angular/core';
+import {
+  FormBuilder,
+  FormGroup,
+  ReactiveFormsModule,
+  Validators,
+} from '@angular/forms';
+import { MatButton } from '@angular/material/button';
+import { MatCardModule } from '@angular/material/card';
+import { MatError, MatFormFieldModule } from '@angular/material/form-field';
+import { MatIconModule } from '@angular/material/icon';
+import { MatInput } from '@angular/material/input';
+import { AuthService } from '@app/auth/auth.service';
+import { capitalizeFirstLetter } from '@app/libs/util';
+import { passwordValidator } from '@app/libs/validators';
+import { AvatarModule } from 'ngx-avatars';
+
+@Component({
+  selector: 'app-settings',
+  standalone: true,
+  imports: [
+    AvatarModule,
+    MatButton,
+    MatIconModule,
+    MatCardModule,
+    MatError,
+    MatFormFieldModule,
+    ReactiveFormsModule,
+    MatInput,
+    CommonModule,
+  ],
+  templateUrl: './settings.component.html',
+  styleUrl: './settings.component.scss',
+})
+export class SettingsComponent implements OnInit {
+  public generalInfoForm!: FormGroup;
+  public passwordForm!: FormGroup;
+
+  public isGeneralInfoEditMode = false;
+  public isPasswordEditMode = false;
+
+  constructor(
+    private authService: AuthService,
+    private fb: FormBuilder
+  ) {}
+
+  ngOnInit(): void {
+    this.generalInfoForm = this.fb.group({
+      name: ['', Validators.required],
+      role: ['', Validators.required],
+      email: ['', [Validators.required, Validators.email]],
+      phone: ['', [Validators.required, Validators.pattern(/^\d{10}$/)]],
+    });
+
+    this.generalInfo();
+
+    this.passwordForm = this.fb.group({
+      currentPassword: [
+        '',
+        [Validators.required, Validators.minLength(8), passwordValidator()],
+      ],
+      confirmNewPassword: [
+        '',
+        [Validators.required, Validators.minLength(8), passwordValidator()],
+      ],
+      newPassword: [
+        '',
+        [Validators.required, Validators.minLength(8), passwordValidator()],
+      ],
+      twoFactorAuthentication: [''],
+    });
+
+    this.passwordInfo();
+  }
+
+  private generalInfo() {
+    this.generalInfoForm.patchValue({
+      name: this.authService.loggedInUser()?.user.name,
+      role: capitalizeFirstLetter(this.authService.role() as string),
+      email: this.authService.loggedInUser()?.user.email,
+      phone: '0211113334',
```

Comment:
Shouldn't this be user's phone?

Reply:
Just make it empty string or?

Reply:
This is not the final work. I just wanted you to have the component you asked for